File: tests/fixtures/sample_project_ruby/src/auth/jwt_utils.rb
# JWT signing and signature verification helpers.
class JwtUtils
  # Produce a signed JWT string from a subject claim.
  def self.encode_token(subject)
    sign("sub=" + subject)
  end

  # Verify a JWT signature and return the payload if valid.
  def self.decode_token(token)
    return token[4..] if token.start_with?("jwt.")
    nil
  end

  # Sign a raw payload (internal helper).
  def self.sign(payload)
    "jwt." + payload
  end
end

File: tests/fixtures/sample_project_php/src/Auth/JwtUtils.php
<?php

namespace Acme\Auth;

/** JWT signing and signature verification helpers. */
class JwtUtils
{
    /** Produce a signed JWT string from a subject claim. */
    public static function encodeToken($subject)
    {
        return self::sign("sub=" . $subject);
    }

    /** Verify a JWT signature and return the payload if valid. */
    public static function decodeToken($token)
    {
        if (strpos($token, "jwt.") === 0) {
            return substr($token, 4);
        }
        return null;
    }

    /** Sign a raw payload (internal helper). */
    private static function sign($payload)
    {
        return "jwt." . $payload;
    }
}
